src/librustc_mir/interpret/memory.rs

```diff
@@ -317,10 +321,10 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> Memory<'a, 'mir, 'tcx, M> {
         &self,
         ptr: Pointer<M::PointerTag>,
         size: Size,
-        access: bool
+        check: InboundsCheck,
     ) -> EvalResult<'tcx> {
         // if ptr.offset is in bounds, then so is ptr (because offset checks for overflow)
-        self.check_bounds_ptr(ptr.offset(size, &*self)?, access)
```

Comment:
This is unrelated to this PR, but I don't think we have a test for the overflow behaviour here.

Reply:
miri tests some things, not sure if it tests this. Also I do not have a good idea for how to test this in a platform-independent way.

But `ptr.offset` certainly fails on overflow. If rustc had unit tests, we'd just let them cover that...